test/Unit/Durable/OrchestrationInvokerTests.cs

```diff
@@ -122,6 +122,18 @@ public void ReturnsNoInvocationOutputWhenStopped()
             Assert.Null(returnOrchestrationMessage.Output);
         }
 
+        [Fact]
+        public void WrapsExceptionIntoOrchestrationFailureException()
+        {
+            Exception originalException = new Exception("Original exception");
+            _mockPowerShellServices.Setup(_ => _.EndInvoke(It.IsAny<IAsyncResult>())).Throws(originalException);
+
+            var output = new[] { "item1", "item2" };
+
+            var thrownException = Assert.Throws<OrchestrationFailureException>(() => InvokeOrchestration(completed: true, output));
+            Assert.Same(originalException, thrownException.InnerException);
+        }
+
```

Comment:
Should we have a test case that validates that the `FullyQualifiedErrorId` contains `Functions.Durable.ActivityFailure`? I can see this becoming a popular search online later on.

Reply:
Good point. We cannot validate this right here because `FullyQualifiedErrorId` does not belong to the exception. It's the other way around: the exception is wrapped into an `ErrorRecord`, and this `ErrorRecord` contains `FullyQualifiedErrorId`... But I agree we should emphasize this because this errorId will become a part of the public contract that we shouldn't break in the future. I think this check should go to `DurableActivityErrorHandler` unit tests, I'll do that.

Reply:
Sounds good. Thanks Anatoli.